### PR TITLE
 🌮 🌮 ErgoTaco Support  🌮 🌮

### PR DESCRIPTION
**Pull request by germ**
Support for the ErgoTaco by g Heavy Industries.

Includes info.json for config.qmk.fm.

### CODE REVIEW

File: keyboards/ergotaco/ergotaco.h [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+#define GPIOA           0x12            // general purpose i/o port register (write modifies OLAT)
+#define GPIOB           0x13
+#define OLATA           0x14            // output latch register
+#define OLATB           0x15
+
+void init_ergodox(void);
+uint8_t init_mcp23018(void);
+
+/*  ---------- LEFT HAND -----------   ---------- RIGHT HAND ---------- */
+#define LAYOUT_TACO(                                                    \
```
Comment (noroadsleft) [inline keyboards/ergotaco/ergotaco.h]: This should be `LAYOUT` as there's only one layout supported.

```suggestion
#define LAYOUT(                                                          \
```

File: keyboards/ergotaco/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ * time to read the QMK docs, a list of keycodes can be found at
+ *
+ * https://github.com/qmk/qmk_firmware/blob/master/docs/keycodes.md
+ *
+ * There's also a template for adding new layers at the bottom of this file!
+ */
+
+#include QMK_KEYBOARD_H
+
+#define IGNORE_MOD_TAP_INTERRUPT
```
Comment (noroadsleft) [inline keyboards/ergotaco/keymaps/default/keymap.c]: Already defined at the keyboard level in `config.h`, and to my knowledge, defining it here wouldn't work under any circumstance.

```suggestion
```

File: keyboards/ergotaco/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+#define TACOTIME 1 // symbols
+
+// Blank template at the bottom
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+/* Keymap template 
+ *
+ * ,-------------------------------------------------.           ,--------------------------------------------.
+ * |       |      |      |      |      |      |      |           |       |      |      |      |      |        |
+ * `-------+------+------+------+------+-------------'           `-------+------+------+------+------+--------' */
+[FIESTA] = LAYOUT_TACO(
```
Comment (noroadsleft) [inline keyboards/ergotaco/keymaps/default/keymap.c]:
```suggestion
[FIESTA] = LAYOUT(  
```

File: keyboards/ergotaco/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+   KC_Q, KC_W, KC_E, KC_R, KC_T, KC_Y,        KC_A, KC_S, KC_D, KC_F, KC_G, KC_H
+),
+};
+
+/* Keymap template 
+ *
+ * ,-------------------------------------------------.           ,--------------------------------------------.
+ * |       |      |      |      |      |      |      |           |       |      |      |      |      |        |
+ * `-------+------+------+------+------+-------------'           `-------+------+------+------+------+--------'
+[FIESTA] = LAYOUT_TACO(
```
Comment (noroadsleft) [inline keyboards/ergotaco/keymaps/default/keymap.c]:
```suggestion
[FIESTA] = LAYOUT(  
```

File: keyboards/ergotaco/keymaps/default/rules.mk [hunk 5]
```diff
@@ -0,0 +1,19 @@
+#----------------------------------------------------------------------------
+# make gergo:germ:dfu
```
Comment (noroadsleft) [inline keyboards/ergotaco/keymaps/default/rules.mk]:
```suggestion
# make ergotaco:default:dfu
```

File: keyboards/ergotaco/keymaps/default/rules.mk [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+#Debug options
+VERBOSE 		 		= yes
+DEBUG_MATRIX_SCAN_RATE  = no
+DEBUG_MATRIX		 	= yes
+
+# A bunch of stuff that you shouldn't touch unless you
+# know what you're doing.
+#
+# No touchy, capiche?
+SRC += matrix.c i2c_master.c
```
Comment (noroadsleft) [inline keyboards/ergotaco/keymaps/default/rules.mk]: This line is already defined at the keyboard level `rules.mk`; it's redundant here.

```suggestion
```

I'll listen if you have a compelling argument for having it at both levels, though.

File: keyboards/ergotaco/readme.md [hunk 7]
```diff
@@ -0,0 +1,22 @@
+# Gergo
+
+![ErgoTaco](https://i.redd.it/dbcu5i21m3i21.jpg)
+
+It's a Taco with a side of HASL. A 12 key fiesta of a board.
+
+Keyboard Maintainer: [Jeremy Bernhardt](https://github.com/germ)  
+Hardware Supported: ErgoTaco
```
Comment (noroadsleft) [inline keyboards/ergotaco/readme.md]: Need two spaces at the end of this line so GitHub renders it properly.

```suggestion
Hardware Supported: ErgoTaco  
```

File: keyboards/ergotaco/readme.md [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+Hardware Availability: [gboards.ca](http://gboards.ca)  
+
+## Firmware building
+After cloning the QMK repo and installing dfu-programmer build and flash with. Be sure to reset your keyboard!
+
+    make ergotaco:default:dfu
+
+To just test your build with the default keymap
+   
+    make gergo:default
```
Comment (noroadsleft) [inline keyboards/ergotaco/readme.md]:
```suggestion
    make ergotaco:default
```

File: keyboards/ergotaco/readme.md [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+![ErgoTaco](https://i.redd.it/dbcu5i21m3i21.jpg)
+
+It's a Taco with a side of HASL. A 12 key fiesta of a board.
+
+Keyboard Maintainer: [Jeremy Bernhardt](https://github.com/germ)  
+Hardware Supported: ErgoTaco
+Hardware Availability: [gboards.ca](http://gboards.ca)  
+
+## Firmware building
+After cloning the QMK repo and installing dfu-programmer build and flash with. Be sure to reset your keyboard!
```
Comment (noroadsleft) [inline keyboards/ergotaco/readme.md]: I think Lines 12-14 would be better as:

```
After cloning the QMK repo and installing dfu-programmer, build and flash with:

    make ergotaco:default:dfu

Be sure to reset your keyboard!

File: keyboards/ergotaco/rules.mk [hunk 10]
```diff
@@ -0,0 +1,19 @@
+#----------------------------------------------------------------------------
+# make gergo:germ:dfu
```
Comment (noroadsleft) [inline keyboards/ergotaco/rules.mk]:
```suggestion
# make ergotaco:default:dfu
```

File: keyboards/ergotaco/keymaps/default/rules.mk (hunk 6 above)
[the remark this answers is not in the document]
Comment (germ) [reply keyboards/ergotaco/keymaps/default/rules.mk]: Moved it around as per your suggestio

**Comment by germ** [comment]
@noroadsleft Some of your commits couldn't be applied because I did them out of order. If you got a second mind ripping over her again?

File: keyboards/ergotaco/readme.md (hunk 9 above)
Comment (noroadsleft) [reply keyboards/ergotaco/readme.md]: The first sentence here still bugs me, with its improper grammar.

\- @noroadsleft 
*son of an English major*

File: keyboards/ergotaco/config.h [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+/* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
+#define LOCKING_SUPPORT_ENABLE
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
+
+/* key combination for command */
+#define IS_COMMAND() ( \
+    keyboard_report->mods == (MOD_BIT(KC_LCTL) | MOD_BIT(KC_RCTL)) || \
+    keyboard_report->mods == (MOD_BIT(KC_LSFT) | MOD_BIT(KC_RSFT)) \
+)
```
Comment (vomindoraan) [inline keyboards/ergotaco/config.h]: We no longer use `keyboard_report->mods` in `IS_COMMAND()`. Also, there are predefined constants for common mod combinations. With that in mind, this define can be replaced with:
```c
#define IS_COMMAND() (get_mods() == MOD_MASK_CTRL || get_mods() == MOD_MASK_SHIFT)
```

**Comment by PyrooL** [comment]
blessed pull request

File: keyboards/ergotaco/config.h (hunk 11 above)
Comment (germ) [reply keyboards/ergotaco/config.h]: Changed in https://github.com/qmk/qmk_firmware/pull/5504/commits/b9b66acb6d4351c72306781afb9314eb4b8e64bc

File: keyboards/ergotaco/rules.mk [hunk 12]
```diff
@@ -0,0 +1,28 @@
+#----------------------------------------------------------------------------
+# make ergotaco:default:dfu
+# Make sure you have dfu-programmer installed!
+# Do not edit this file! Make a copy of keymaps/default and modify that!
+#----------------------------------------------------------------------------
+# Source includes
+SRC += matrix.c i2c_master.c
```
[the remark this answers is not in the document]
Comment (germ) [reply keyboards/ergotaco/rules.mk]: Fixed in https://github.com/qmk/qmk_firmware/pull/5504/commits/1a305b5b4e92d509306e8446d01d7ef6ad048435

**Comment by germ** [comment]
What else?

image

**Comment by drashna** [comment]
Meme-tacular! 

This looks good to go.  Though.... I kind of want to wait until #5498 is merged, as this keyboard looks to be based on the ergodox code.   It's an update to the debouncing algorithm, mostly.   

If you would like me us to merge it now, and maybe update the code later, that's perfectly fine, and we'll do it.

**Comment by germ** [comment]
Go ahead and merge it now, I'll update the code later :D

Thanks Drash!

**Comment by drashna** [comment]
WIll do, then!

**Pull request merged**